### PR TITLE
Added suggestion not to use synced folder for volume directory

### PR DESCRIPTION
**Pull request by soltysh**
@bparees I've added the suggestion to use different dir for volume-dir, is that enough or I should write something more about it?

### CODE REVIEW

File: examples/sample-app/README.md [hunk 1]
```diff
@@ -121,6 +121,8 @@ This section covers how to perform all the steps of building, deploying, and upd
 
     Note: sudo is required so the kubernetes proxy can manipulate iptables rules to expose service ports.
 
+    Note: when using vagrant synced folder it is advised to use different directory than the one in synced folder. This can be achieved by passing `--volume-dir=/absolute/path` to `openshift start` command.
```
Comment (bparees) [inline examples/sample-app/README.md]: "it is advices to use a different directory for volume storage rather than using the synced folder."

**Comment by smarterclayton** [comment]
This really belongs in the setup instructions in CONTRIBUTING

**Comment by smarterclayton** [comment]
Please update to be in CONTRIBUTING

**Comment by soltysh** [comment]
Will update along with entire CONTRIBUTING file which is pointing to README which again redirects to CONTRIBUTING for more info on how to run OpenShift in development. I'd like to clear all that out. Hopefully I'll tackle that on Monday.

**Comment by kargakis** [comment]
This shoud stay in sample-app as well

**Comment by soltysh** [comment]
Added to CONTRIBUTING, (with some cleaning as well) and troubleshooting. ptal

File: CONTRIBUTING.adoc [hunk 2]
```diff
@@ -132,15 +132,33 @@ To facilitate rapid development we've put together a Vagrantfile you can use to
...
-        $ make build
+        $ make clean build
 
-7.  Start an OpenShift all-in-one server in SSH (includes everything you need to try OpenShift)
+7.  Now change into the directory with the OpenShift binaries, and start the OpenShift server:
 
-        $ sudo systemctl start openshift
+        $ cd _output/local/go/bin
+        $ sudo ./openshift start
+        $ sudo ./openshift start --public-master=localhost &> logs/openshift.log &
```
Comment (bparees) [inline CONTRIBUTING.adoc]: you've got two starts here.

File: CONTRIBUTING.adoc [hunk 3]
```diff
@@ -132,15 +132,33 @@ To facilitate rapid development we've put together a Vagrantfile you can use to
...
+
++
+NOTE: when using vagrant synced folder it is advised to use different directory than the one in synced folder. This can be achieved by passing `--volume-dir=/absolute/path` to `openshift start` command.
```
Comment (bparees) [inline CONTRIBUTING.adoc]: not being  a heavy vagrant user, is it obvious which is the "synced" folder?

also at a minimum it should be "than the one in *the* synced folder"

Comment (bparees) [inline CONTRIBUTING.adoc]: and "use a different directory for volume storage than the one in the synced folder"

File: examples/sample-app/README.md (hunk 1 above)
Comment (bparees) [inline examples/sample-app/README.md]: comment still applies.

File: docs/debugging-openshift.md [hunk 4]
```diff
@@ -113,6 +113,22 @@ There are a number of suspicious looking messages that appear in the openshift l
...
+
+When using [vagrant synced folder](http://docs.vagrantup.com/v2/synced-folders/) you may encounter
+following errors in OpenShift log:
+
+        E0706 11:29:43.421460    3664 empty_dir.go:322] Expected directory "/data/src/github.com/openshift/origin/openshift.local.volumes/pods/4c390e43-23d2-11e5-b42d-080027c5bfa9/volumes/kubernetes.io~secret/deployer-token-f9mi7" permissions to be: -rwxrwxrwx; got: -rwxrwxr-x
+        E0706 11:29:43.421741    3664 kubelet.go:1114] Unable to mount volumes for pod "docker-registry-1-deploy_default": operation not supported; skipping pod
+        E0706 11:29:43.438449    3664 pod_workers.go:108] Error syncing pod 4c390e43-23d2-11e5-b42d-080027c5bfa9, skipping: operation not supported
+
+This will happen when using our provided Vagrantfile to develop OpenShift with vagrant. One of the reasons
+is that you can't use ACLs on shared directories. The solution to this problem is using different directory
```
Comment (bparees) [inline docs/debugging-openshift.md]: "The solution to this problem is to use a different directory for volume storage than the one in the synced folder"

File: CONTRIBUTING.adoc (hunk 3 above)
Comment (soltysh) [inline CONTRIBUTING.adoc]: I'll add info about us doing that for you.

**Comment by soltysh** [comment]
@bparees comments applied, thx.

**Comment by bparees** [comment]
lgtm.

**Comment by soltysh** [comment]
@smarterclayton any objects in merging this one?

**Comment by smarterclayton** [comment]
[merge]

**Pull request merged**